src/org/pahappa/systems/registrationapp/views/UserView.java

```diff
@@ -133,35 +188,46 @@ private void getUserOfUsername() {
             System.out.println("First Name: " + user.getFirstname());
             System.out.println("Last Name: " + user.getLastname());
             System.out.println("Username: " + user.getUsername());
-            System.out.println("Date Of Birth: " + user.getDateOfBirth());
+            System.out.println("Date Of Birth: " + user.getDateOfBirth().toString().replaceFirst("00:00:00 UTC ", ""));
 
         } catch (Exception e) {
-            System.err.println("An error has been encountered with your inputs. Please hit r to try again.");
-            String retry = scanner.nextLine().strip();
-            if (retry.toLowerCase().equals("r")) {
-                getUserOfUsername();
-            }
+            if (retry()) getUserOfUsername();
```

Comment:
The catch block can be improved to provide more specific error handling message especially before retrying. Something like this
catch (Exception e) {
        System.out.println("An error occurred while retrieving the user. Please try again.");
        if (retry()) {
            getUserOfUsername();
        }
    }